src/tracker_manager.cpp

```diff
@@ -286,10 +294,34 @@ constexpr tracker_request_flags_t tracker_request::i2p;
 			m_udp_conns[con->transaction_id()] = con;
 			con->start();
 			return;
-		}
-
+        }
+#if TORRENT_USE_RTC
+        else if (protocol == "ws" || protocol == "wss")
+        {
+			std::shared_ptr<request_callback> cb = c.lock();
+			if(!cb) return;
+			const int max_offers = 10;
+			req.num_want = std::min(req.num_want, max_offers);
+			cb->generate_rtc_offers(req.num_want
+				, [this, &ios, req = std::move(req), c](error_code const& ec
+					, std::vector<aux::rtc_offer> const& offers) mutable
+			{
+				if(!ec) req.offers = offers;
+
+				auto it = m_websocket_conns.find(req.url);
```

Comment:
Good point. The URL should be the exact same since it will be the one from the torrent file in the successive requests. The only situation where it could be different is because there are two different tracker URLs in the torrent file.